core/src/test/java/brooklyn/entity/rebind/RebindCatalogItemTest.java

```diff
@@ -192,6 +194,23 @@ public void testSameCatalogItemIdRemovalAndAdditionRebinds() throws Exception {
         rebindAndAssertCatalogsAreEqual();
     }
 
+    @Test
+    public void testRebindAfterItemDeprecated() {
+        String yaml =
+                "name: rebind-yaml-catalog-item-test\n" +
+                "brooklyn.catalog:\n" +
+                "  version: " + TEST_VERSION + "\n" +
+                "services:\n" +
+                "- type: io.camp.mock:AppServer";
+        BasicBrooklynCatalog catalog = (BasicBrooklynCatalog) origManagementContext.getCatalog();
+        CatalogItem<?, ?> catalogItem = catalog.addItem(yaml);
+        assertNotNull(catalogItem, "catalogItem");
+        catalogItem.setDeprecated(true);
```

Comment:
It was passing because after rebinding I was getting `catalogItemAfterRebind` from `catalog`. I've changed it to `CatalogItem<?, ?> catalogItemAfterRebind = newManagementContext.getCatalog().getCatalogItem("rebind-yaml-catalog-item-test", TEST_VERSION);` and it's failing without the catalog.persist(catalogItem)